### PR TITLE
Per-stage shader keywords in HDRP

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
With this PR HDRP will use per-stage keywords what that means is that we can do stuff like `multi_compile_fragment` to have only the fragment shader variant computed. It is very important to define one separate line per stage that needs to include the keyword. This work for both multi compiles and shader_features 

This is has to be done manually for code shaders (I did a pass on our shaders, being conservative to be safe) and I added a field to have that easily generated in SG. 

This reduces the number of compiled shader programs with all the benefits that come from that. 

I ran test manually and they pass.

### CODE REVIEW

File: com.unity.shadergraph/Editor/Data/Util/KeywordUtil.cs [hunk 1]
```diff
@@ -65,11 +66,57 @@ public static string ToDeclarationString(this KeywordDefinition keywordDefinitio
...
+            if ((keyword.stages & KeywordShaderStage.Geometry) == KeywordShaderStage.Geometry)
+                dstStages.Add(KeywordShaderStage.Geometry);
+            if ((keyword.stages & KeywordShaderStage.RayTracing) == KeywordShaderStage.RayTracing)
+                dstStages.Add(KeywordShaderStage.RayTracing);
+
+            return dstStages.Count > 1;
+
+        }
+
+        // TODO: BROKEN FIX!!!!
```
Comment (alelievr) [inline com.unity.shadergraph/Editor/Data/Util/KeywordUtil.cs:92]: what is broken? :)

**Review by alelievr — APPROVED** [review]
LGTM!

(with the TODO fixed)

File: com.unity.shadergraph/Editor/Data/Util/KeywordUtil.cs (hunk 1 above)
Comment (FrancescoC-unity) [reply com.unity.shadergraph/Editor/Data/Util/KeywordUtil.cs:92]: Oops this escaped my cleanup :D Will remove immediately!

**Review by cdxntchou — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
Adding QA: QA please try to load a project like SpaceShip or Palace, and build a player a check that everything is fine. Thanks

**Review by iM0ve — APPROVED** [review]
I have upgraded Amalienborg Palace project and also built a Player. Did not notice any visual issues or any new console errors

image

**Review by sebastienlagarde — APPROVED** [review]

**Review by LandonTownsendUnity — APPROVED** [review]
Just made sure that URP still works properly when HDRP is in project.

**Pull request merged**